src/jdk.compiler/share/classes/com/sun/tools/javac/file/JavacFileManager.java

```diff
@@ -571,7 +576,10 @@ public ArchiveContainer(Path archivePath) throws IOException, ProviderNotFoundEx
                     throw new IOException("ZipException opening \"" + archivePath.getFileName() + "\": " + ze.getMessage(), ze);
                 }
             } else {
-                this.fileSystem = FileSystems.newFileSystem(archivePath, (ClassLoader)null);
+                // Less common case is possible if the file manager was not initialized in JavacTask,
+                // or if non "*.jar" files are on the classpath. At the time of writing, both `javac -cp a.zip`
+                // and `javac -cp x.JAR` file would hit this branch which may warrant investigation.
```

Comment:
Hello Jason, do we need that second sentence to be added as a code comment? I feel it creates confusion on what's expected to be done. Perhaps just add that first sentence and remove the second?

Reply:
Could you also update the copyright year on this file from 2024 to 2025?

Reply:
All done now.